src/Microsoft.DotNet.Web.ProjectTemplates/content/StarterWeb-CSharp/Areas/Identity/Controllers/AccountController.cs

```diff
@@ -327,6 +327,8 @@ public IActionResult ExternalLogin(string provider, string returnUrl = null)
                         return RedirectToLocal(returnUrl);
                     }
                 }
+
+                ViewData["LoginProvider"] = info.LoginProvider;
```

Comment:
Should this line go right before line 335 instead? That is, if ViewState is not valid, doesn't it also need this to be set?

I think you could test it out like so:
1. Disable JavaScript in the browser (easy to do using Chrome dev tools)
2. Navigate to the preceding page
3. Have a validation error in the form (whatever the form is)
4. Click the button that goes to this page

Reply:
And the same question applies to the Razor Pages code above.